### PR TITLE
Snowflake ingestion fix

### PR DESCRIPTION
**Pull request by codingwithabhi**
### Describe your changes :

I worked on the fixing ingestion fail issue and variant data type ingestion issue in snowflake

#
### Type of change :

- [x] Bug fix
- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@ayush-shah

### CODE REVIEW

**Comment by ayush-shah** [comment]
@codingwithabhi let's test and check other connectors as well, so that we are sure that it's not a breaking change

File: ingestion/src/metadata/utils/column_type_parser.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def get_column_type(column_type: Any) -> str:
         if isinstance(column_type, types.NullType):
             return "NULL"
         for sql_type in ColumnTypeParser._SOURCE_TYPE_TO_OM_TYPE.keys():
-            if str(column_type) in sql_type:
+            if str(column_type) == sql_type:
```
Comment (harshach) [inline ingestion/src/metadata/utils/column_type_parser.py:149]: @ayush-shah is this the same problem in hive connector as well

Comment (ayush-shah) [reply ingestion/src/metadata/utils/column_type_parser.py:149]: @harshach no, this is not the same as hive, as we are checking it with the _SOURCE_TYPE_TO_OM_TYPE whereas in the hive, we are testing hive out with _COLUMN_TYPE_MAPPING.

**Comment by harshach** [comment]
@codingwithabhi @ayush-shah what is pending on this PR?

**Comment by ayush-shah** [comment]
I think it's ready to be merged in.
@codingwithabhi can you open this PR for review?

**Review by harshach — APPROVED** [review]

**Pull request merged**